iset.mm

```diff
@@ -104463,6 +104596,106 @@ which defines the set we are taking the supremum of must (a) be true at
+      pm5.21nii rexeqi imbi2i ralbii anbi12i rexbii ) ADUEZCUEZUMOZUFZDUAUEZUGZ
+      NUEZUGZGPZNQUHZPZUAQUHZUIZUUTUUSUMOZEUEZUUSUMOZEUVJUJZUKZDQUIZRZCQUJUVBDG
+      UIZUVLUVNEGUJZUKZDQUIZRZCQUJACDEUAUVHSQUNAUVLUFZDUVHUIZUVAUUSUVMUMOZEUVHU
+      JZUKZDQUIZRZCSUJZUWJCQUJULAUWDDUVGNSUHZUIZUVAUWFEUWLUJZUKZDQUIZRZCSUJUWKA
+      UVGFUGZUGZGPCDEUBNUWRAFAFIUOVEZPZFSPZAFBHUWTUHZPZUXAAFGPUXDLGUXCFKUPUQBHF
+      UWTURTZIFUSTZUTZUVEUWRVAUVFUWSGUVEUWRVBVCAUWSFGAFAFUXFVFVDLVGAUVEUWRUOVEZ
+      PZRZIUVFVHOZUVGVIZUXKUFZUXJUXKRZUXLBHUVFVJZVIZUXNUVFIFVKVLZPZBVIZHUXQUIZU
+      XPUXNUXRUXKUVFFVHOZRZUXNUXKUYAUXJUXKVRUXJUYAUXKUXJFUVEUXJFAUXBUXIUXFVMVNU
+      XJUVEUXIUVESPZAUWRUVEUSVOZVNUXIUWRUVEVHOAUWRUVEVPVOVQVMVSUXNUVFSPZISPZUXB
+      UXRUYBVTUXJUYEUXKUXJUVEUYDUTZVMAUYFUXIUXKJWAAUXBUXIUXKUXFWAUVFIFWBWCWDZAU
+      XTUXIUXKAUXSHUXQMWEWAUXSUXPHUVFUXQUXOHBHUVFWFWIHUEUVFVABUXOBHUVFWGWHWJWKU
+      XNUVGUXOUVGUVFUXCPZUXNUXOGUXCUVFKUPUXNUXOUVFUWTPZUXORUYIUXNUYJUXOUXNUXRUY
+      JUYHUVFIFWLTWMBHUVFUWTHUWTWNWOWPWQWHWDUXJUXMRZUVGUVGUFZWRUXLUYKUYLUVGUYKU
+      XKUVGUXJUXMVRUXKUVFUXCGUYIUYJUXKBHUVFUWTURZIUVFVPZTKXAWSWTUVGXBXCUXJUXKVI
+      ZUXKUXMWRUXJUYFUYEUYOAUYFUXIJVMUYGIUVFXDXEUXKXLTXFAIUGZXGXHVLZUXHPZUYLNUY
+      QUOVEZUIZUYLNUBUEZUOVEZUIZUBUXHUJAUYPUXHPZUYRAVUDUWRUYPVHOZAIFVHOZVUEAUXA
+      VUFUXEIFVPTAIFAIJVNZAFUXFVNXIXJAUWRSPUYPSPZVUDVUEVTUXGAIJUTZUWRUYPYAXEWDU
+      WRUYPXKTAUYLNUYSAUVEUYSPZRZUVGUYPUYPUMOZVUKUVGRZVULUYQUYPVHOZVUMUYQUVEUYP
+      VUMUYPQPUYQQPVUMIAIQPVUJUVGVUGWAZXMZUYPXNTVUMUVEVUJUYCAUVGUYQUVEUSXOVNZVU
+      PVUJUYQUVEVHOAUVGUYQUVEVPXOVUMIUVEVUOVUQAUVGUXKVUJAUVGRUYJUXKUVGUYJAUYJUV
+      FUXCGUYMKXAVOUYNTXPXQXRVUMVUHVUHVULVUNVTAVUHVUJUVGVUIWAZVURUYPUYPXSXEWDVU
+      MUYPVUPXTYBWEVUCUYTUBUYQUXHVUAUYQVAUYLNVUBUYSVUAUYQUOYCYDYEXEYFAUWQUWJCSA
+      UWMUWEUWPUWIAUWDDUWLUVHAUCUWLUVHUCUEZUWLPZVUSUVHPZVTAVUSSPZVUSUGZGPZRZVUS
+      QPZVVDRZVUTVVAVVEVVGVVBVVFVVDVUSYGYHVVGVVBVVDVVGVVBVVCSPZVVDVVHVVFVVDVVCU
+      WTPZVVHVVIVVCUXCGBHVVCUWTURKXAIVVCUSTVOVVFVVBVVHVTZVVDVVFVUSYIPVVJVUSYJVU
+      SYKTVMWDVVFVVDVRVSYLUVGVVDNVUSSNUCYMUVFVVCGUVEVUSVBVCZYNUVGVVDNVUSQVVKYNY
+      OYPYQZYDAUWOUWHDQAUWNUWGUVAAUWFEUWLUVHVVLYRYSYTUUAUUBXJUWJCSQUUCUUDUVHQUN
+      AUVGNQUUEYPUUFUVRUWCCQUVKUVSUVQUWBUVBDUVJGUDUVJGUDUEZUVJPZVVMQPZVVMGPZUVI
+      UAVVMQURVVPVVMUWTPZVVOVVQVVMUXCGBHVVMUWTURKXAIVVMUUGTVVOVVNVVMUGZUVHPZVVR
+      UGZGPZVVPUVIVVSUAVVMQUAUDYMUVDVVRUVHUVCVVMVBVCUUHVVOVWAVVRQPZVWARVVSVVOVW
+      BVWAVVMUUIWMUVGVWANVVRQUVEVVRVAUVFVVTGUVEVVRVBVCYNWPVVOVVTVVMGVVOVVMVVMYJ
+      VDVCUUJUUMUUKZUULUVPUWADQUVOUVTUVLUVNEUVJGVWCUUNUUOUUPUUQUURUQ $.
+
+    $d A j m y $.  $d A m n $.  $d M j m $.  $d M m n $.  $d S a b m x y z $.
+    $d S j m x y z $.  $d S a m w x y z $.  $d a b m n ph $.
+    $d j m ph x y z $.
+    $( The infimum of a subset of an upper set of integers is less than or
+       equal to all members of the subset.  (Contributed by Jim Kingdon,
+       13-Jan-2022.) $)
+    infssuzledc $p |- ( ph -> inf ( S , RR , < ) <_ A ) $=
```

Comment:
I was wondering why this was not trivial in much greater generality... And this is, seemingly, because df-sup is defined in terms of `-. R` ?  I think I made a proposal a few commits ago about defining it in a more streamlined way... I'll have to get back to it.

Reply:
Maybe I need to see what you are thinking in the form of proofs? This is just a translation of http://us.metamath.org/mpeuni/infssuzle.html with a bit of decidability added. So it isn't really more or less trivial than set.mm except for the decidability (which set.mm gets for free).

Reply:
I mean changing df-sup, although I cannot retrieve my proposal. Currently, we have
> sup ( A , B , R ) = U. { x e. B | ( A. y e. A -. x R y /\ A. y e. B ( y R x -> E. z e. A y R z ) ) }

and I meant replacing -. x R y with ( x R y \\/ x = y ) , which seems to translate better "upper bound" and avoids negation is more faithful in intuitionistic logic.

Reply:
> I meant replacing -. x R y with ( y R x \/ y = x )

I'm a little unsure of what to tell you on this beyond "give it a try and see how disjunction and equality behave when you try to write proofs using them". Maybe try to prove something like maxle1 using this definition. Seems like this could be fruitful whether the result is "you discover something that Mario and I have been missing" or "you better understand what we are talking about" (or some of each, even).